clean repo and add editorconfig file

FiF0o opened the pull request:
I have noticed that the .gitignore file could be simplified and cleaned some of the project settings not needed - .idea/
The history and cached files for .idea have been clean as well.

Glennmen commented:
@FiF0o I really like this PR, could you tell me if your .editorconfig complies php PSR2 linting?
I was thinking about making it standard for this project.

FiF0o commented:
Thanks @Glennmen,
I believe it does, I have added a few more rules to enforce PSR2.
As I am not a php expert you should check whether it enforces the rules or not.
I have changed the committed files and added a few more lines.
```
charset = utf-8
max_line_length = off
indent_brace_style = Allman
spaces_around_brackets = none
```

Just for reference the list of the properties can be found here: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties
